[BUG] CMSIS_6 module required by the ARM port for Cortex-M

**Describe the bug**
The Build pipeline is broken since about one week. It seems related to Zephyr upstream issue zephyrproject-rtos/zephyr#89951 and pull-request zephyrproject-rtos/zephyr#89370.

**To Reproduce**
Steps to reproduce the behavior:
1. use latest Bridle main branch
2. `west build -b magpie_f777ni -p -d build/led_blinky-magpie_f777ni zephyr/samples/basic/blinky`
3. See error (below)

**Expected behavior**
No warnings, no errors about missing core/arch components.

**Impact**
Showstopper, Bridle's QA have to run w/o errors or warnings.

**Logs and console output**
```
[10/152] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
FAILED: zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj 
ccache /opt/zephyr-sdk-0.17.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCORE_CM7 -DHSE_VALUE=25000000 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DPICOLIBC_LONG_LONG_PRINTF_SCANF -DSTM32F777xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I/.../workspace/zephyr/kernel/include -I/.../workspace/zephyr/arch/arm/include -I/.../workspace/build/led_blinky-magpie_f777ni/zephyr/include/generated/zephyr -I/.../workspace/zephyr/include -I/.../workspace/build/led_blinky-magpie_f777ni/zephyr/include/generated -I/.../workspace/bridle/soc/st/stm32 -I/.../workspace/zephyr/soc/st/stm32/common/. -I/.../workspace/zephyr/drivers -I/.../workspace/zephyr/soc/st/stm32/stm32f7x/. -I/.../workspace/bridle/include -I/.../workspace/bridle/lib/bridle/include -I/.../workspace/modules/hal/stm32/stm32cube/stm32f7xx/soc -I/.../workspace/modules/hal/stm32/stm32cube/stm32f7xx/drivers/include -I/.../workspace/modules/hal/stm32/stm32cube/common_ll/include -isystem /.../workspace/zephyr/lib/libc/common/include -fno-strict-aliasing -Os -imacros /.../workspace/build/led_blinky-magpie_f777ni/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/opt/zephyr-sdk-0.17.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros /.../workspace/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/.../workspace/zephyr/samples/basic/blinky=CMAKE_SOURCE_DIR -fmacro-prefix-map=/.../workspace/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/.../workspace=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c99 -fno-lto -MD -MT zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -c /.../workspace/zephyr/arch/arm/core/offsets/offsets.c
In file included from /.../workspace/zephyr/include/zephyr/arch/arm/asm_inline.h:18,
                 from /.../workspace/zephyr/include/zephyr/arch/arm/arch.h:30,
                 from /.../workspace/zephyr/include/zephyr/arch/cpu.h:19,
                 from /.../workspace/zephyr/include/zephyr/kernel_includes.h:36,
                 from /.../workspace/zephyr/include/zephyr/kernel.h:17,
                 from /.../workspace/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from /.../workspace/zephyr/arch/arm/core/offsets/offsets.c:9:
/.../workspace/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:24:10: fatal error: cmsis_core.h: No such file or directory
   24 | #include <cmsis_core.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /.../workspace/build/led_blinky-magpie_f777ni
```

**Screenshots**
Non.

**Development Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.17.0
 - Commit SHA or Version: c1ca724db887c9ce38b630f2b932568803860387 (main)

**Additional context**
Non.

--- a/doc/bridle/releases/release-notes-4.2.0.rst
+++ b/doc/bridle/releases/release-notes-4.2.0.rst
@@ -217,6 +217,7 @@ Issue Related Items
 
 These GitHub issues were addressed since project bootstrapping:
 
+* :github:`320` - [BUG] CMSIS_6 module required by the ARM port for Cortex-M
 * :github:`317` - [BUG] Rename Kconfig option ``SCHED_DUMB`` and ``WAITQ_DUMB``
 * :github:`316` - [BUG] Remove Kconfig option ``ETH_STM32_HAL_MII`` and ``ETH_STM32_HAL_PHY_ADDRESS``
 * :github:`310` - [HW] STK8BA58 accelerometer

--- a/doc/bridle/shortcuts.txt
+++ b/doc/bridle/shortcuts.txt
@@ -732,9 +732,6 @@
 .. |zephyr:board:qemu_malta| replace::
    :external+zephyr:zephyr:board:`qemu_malta`
 
-.. |zephyr:board:qemu_nios2| replace::
-   :external+zephyr:zephyr:board:`qemu_nios2`
-
 .. |zephyr:board:qemu_riscv32e| replace::
    :external+zephyr:zephyr:board:`qemu_riscv32e`
 

--- a/samples/helloshell/README.rst
+++ b/samples/helloshell/README.rst
@@ -36,7 +36,6 @@ Requirements
   * |zephyr:board:qemu_riscv64|
   * |zephyr:board:qemu_xtensa|
   * |zephyr:board:qemu_leon3|
-  * |zephyr:board:qemu_nios2|
   * |zephyr:board:nucleo_f303re| (NUCLEO-F303RE)
   * |zephyr:board:nucleo_f401re| (NUCLEO-F401RE)
   * |zephyr:board:nucleo_f413zh| (NUCLEO-F413ZH)

--- a/samples/helloshell/sample.yaml
+++ b/samples/helloshell/sample.yaml
@@ -51,7 +51,6 @@ common:
     - qemu_xtensa/dc233c
     - qemu_xtensa/dc233c/mmu
     - qemu_leon3
-    - qemu_nios2
     - arduino_zero
     - cytron_maker_nano_rp2040
     - cytron_maker_pi_rp2040

--- a/submanifests/zephyr.yml
+++ b/submanifests/zephyr.yml
@@ -53,6 +53,7 @@ manifest:
           - canopennode
           - chre
           - cmsis
+          - cmsis_6
           - edtt
           - fatfs
           - hal_adi

--- a/tests/bridle/common/testcase.yaml
+++ b/tests/bridle/common/testcase.yaml
@@ -46,7 +46,6 @@ common:
     - qemu_xtensa/dc233c
     - qemu_xtensa/dc233c/mmu
     - qemu_leon3
-    - qemu_nios2
 tests:
   bridle.common:
     build_on_all: true

